openupgrade_scripts/scripts/stock/tests/test_migration.py

```diff
@@ -0,0 +1,135 @@
+from odoo.tests import TransactionCase
+
+from odoo.addons.openupgrade_framework import openupgrade_test
+
+
+@openupgrade_test
+class TestStockMigration(TransactionCase):
+    def test_picking_type_required_fields(self):
+        """Test that newly required fields are set"""
+        for picking_type in self.env["stock.picking.type"].search([]):
+            self.assertTrue(picking_type.default_location_src_id)
+            self.assertTrue(picking_type.default_location_dest_id)
+
+    def test_pull_moves(self):
+        """
+        Test that pull moves have been migrated correctly and new moves yield the
+        same result
+        """
+        product = self.env.ref("openupgrade_test_stock.pull_product")
+        stock_location = self.env.ref("stock.stock_location_stock")
+        intermediate_location = self.env.ref(
+            "openupgrade_test_stock.intermediate_pull_location"
+        )
+        customer_location = self.env.ref("stock.stock_location_customers")
+
+        moves = self.env["stock.move"].search([("product_id", "=", product.id)])
+        from_stock = moves.filtered(lambda x: x.location_id == stock_location)
+        from_intermediate = moves.filtered(
+            lambda x: x.location_id == intermediate_location
+        )
+
+        self.assertEqual(from_stock.location_dest_id, intermediate_location)
+        self.assertEqual(from_stock.location_final_id, intermediate_location)
+
+        self.assertEqual(from_intermediate.location_dest_id, customer_location)
+        self.assertEqual(from_intermediate.location_final_id, customer_location)
+
+        self.env["stock.rule"].search(
+            [
+                "|",
+                ("location_src_id", "=", intermediate_location.id),
+                ("location_dest_id", "=", intermediate_location.id),
+            ]
+        ).location_dest_from_rule = True
```

Comment:
I don't think it should be necessary to set this here, should come out of the migration like that

Reply:
I mean, the migration expects to maintain compatibility. Odoo, in their comments, explain this:

![Selection_4625](https://github.com/user-attachments/assets/e6c31753-9990-425a-bf12-e56727a1c818)
![Selection_4626](https://github.com/user-attachments/assets/af2d7327-7db4-4df3-8509-09fc134fe7ad)

Reply:
but they test its effect. remove the line I'm commenting on, and you get different (wrong) location_dest_ids below